contracts/bid/src/hosted/transaction.rs

```diff
@@ -131,18 +152,27 @@ impl Contract {
         };
         let bid: &mut BidLeaf = branch_mut.deref_mut();
 
+        // Check wether the maturity and expiration periods are within bounds.
+        let block_height = dusk_abi::block_height();
+        let bid_expiration = bid.bid().expiration().clone();
+        // σh + Δmaturity​ ≥ B.Bhexpiration​)
+        assert!(block_height + MATURITY_PERIOD >= bid_expiration);
+        // B.Bhexpiration​ > σh
+        assert!(bid_expiration > block_height);
+
         // Verify schnorr sig.
         if !rusk_abi::verify_schnorr_sign(
             sig,
             pk,
-            BlsScalar::from(*bid.0.expiration()),
+            BlsScalar::from(bid_expiration),
         ) {
             return false;
         }
 
         // Assuming now that the result of the verification is true, we now
-        // should update the expiration of the Bid by `EXPIRATION_PERIOD`.
-        bid.0.extend_expiration(EXPIRATION_PERIOD);
+        // should update the expiration of the Bid by `VALIDITY_PERIOD`.
+        bid.bid_mut().extend_expiration(VALIDITY_PERIOD);
+        bid.expiration_mut().0 += VALIDITY_PERIOD;
```

Comment:
Elaborate better on the comments on why we need to expose `.0` and why we don't have an API to prevent this coupled code

Reply:
I just expose 0 on the crate and it's because I need to access the internals of the Leaf.